internal/store/db/mock/store.go

```diff
@@ -0,0 +1,129 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// mock provides the Mock implementation of the StoreClient interface for unit tests
+package mock
+
+import (
+	"errors"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/contracts"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/db/interfaces"
+
+	"github.com/google/uuid"
+)
+
+// Client provides a wrapper for Mongo's Client type
+type Client struct {
+	store map[string]contracts.StoredObject
+}
+
+// Store persists a stored object to the data store.
+func (c Client) Store(object contracts.StoredObject) (string, error) {
+	err := validateContract(false, object)
+	if err != nil {
+		return "", err
+	}
+
+	if object.ID == "" {
+		object.ID = uuid.New().String()
+	}
+
+	// determine if this object already exists in the store
+	_, ok := c.store[object.ID]
+	if ok {
+		return "", errors.New("object exists in store")
+	}
+
+	c.store[object.ID] = object
+
+	return object.ID, nil
+}
+
+// RetrieveFromStore gets an object from the data store.
+func (c Client) RetrieveFromStore(appServiceKey string) (objects []contracts.StoredObject, err error) {
+
+	for _, object := range c.store {
+		if object.AppServiceKey == appServiceKey {
+			objects = append(objects, object)
+		}
+	}
+
+	return objects, nil
+}
+
+// Update replaces the data currently in the store with the provided data.
+func (c Client) Update(object contracts.StoredObject) error {
+	err := validateContract(true, object)
+	if err != nil {
+		return err
+	}
+
+	item, ok := c.store[object.ID]
+	if !ok || item.AppServiceKey != object.AppServiceKey {
+		return errors.New("item not found")
+	}
+
+	c.store[item.ID] = object
+	return nil
+}
+
+// RemoveFromStore removes an object from the data store.
+func (c Client) RemoveFromStore(object contracts.StoredObject) error {
+	err := validateContract(true, object)
+	if err != nil {
+		return err
+	}
+
+	item, ok := c.store[object.ID]
+	if !ok || item.AppServiceKey != object.AppServiceKey {
+		return errors.New("item not found")
+	}
+
+	delete(c.store, item.ID)
+	return nil
+}
+
+func (c Client) Disconnect() error {
+	return nil
+}
+
+// NewClient provides a factory for building a Mock StoreClient
+func NewClient() (client interfaces.StoreClient, err error) {
+	client = Client{
+		store: make(map[string]contracts.StoredObject),
+	}
+	return client, nil
+}
+
+func validateContract(IDRequired bool, o contracts.StoredObject) error {
```

Comment:
One of the things I did on my Redis branch that I think we should do here too is make this a method on the StoredObject contract. It would be silly to have this reimplemented on every driver.

Reply:
I understand where you are coming from, but it doesn't below on the Interface. What about a common helper module that the common code can reside in?